### PR TITLE
Fix: Issue with scaling of dynamics #18860

### PR DESCRIPTION
**Pull request by mike-spa**
Resolves: #18860

### CODE REVIEW

**Comment by cbjeukendrup** [comment]
Interesting solution; especially a lot simpler than what I did in #15772. However, it might be preferable to specify the style conversion function in the call to `buildPropertyItem`, to keep it nearby the conversion function for setting the property.

**Comment by mike-spa** [comment]
> Interesting solution; especially a lot simpler than what I did in #15772. However, it might be preferable to specify the style conversion function in the call to `buildPropertyItem`, to keep it nearby the conversion function for setting the property.

Well in this case the style conversion function is just the same as the property conversion: it's just *100 or /100. In fact I wondered if there's a way to reuse the same conversion function that we're passing to builPropertyItem, but I couldn't come up with a sensible way

**Review by oktophonie — APPROVED** [review]

File: src/inspector/models/notation/dynamics/dynamicsettingsmodel.cpp [hunk 1]
```diff
@@ -39,8 +39,12 @@ DynamicsSettingsModel::DynamicsSettingsModel(QObject* parent, IElementRepository
 void DynamicsSettingsModel::createProperties()
 {
     m_avoidBarLines = buildPropertyItem(mu::engraving::Pid::AVOID_BARLINES);
-    m_dynamicSize = buildPropertyItem(mu::engraving::Pid::DYNAMICS_SIZE, [this](const mu::engraving::Pid pid, const QVariant& newValue) {
+    m_dynamicSize = buildPropertyItem(mu::engraving::Pid::DYNAMICS_SIZE,
+                                      [this](const mu::engraving::Pid pid, const QVariant& newValue) {
         onPropertyValueChanged(pid, newValue.toDouble() / 100);
+    },
+                                      [this](const mu::engraving::Sid sid, const QVariant& newValue) {
+        updateStyleValue(sid, newValue.toDouble() / 100);
```
Comment (cbjeukendrup) [inline src/inspector/models/notation/dynamics/dynamicsettingsmodel.cpp:47]: In the default style callback, there is also `emit requestReloadPropertyItems();`. Do we need that here too?

**Comment by cbjeukendrup** [comment]
There are some other properties that have similar issues with similar fixes:
- the `MAG` property in FretDiagramSettingsModel
- the `SCALE` property in TimeSignatureSettingsModel

Maybe it'd be nice to add those to this PR.

File: src/inspector/models/notation/timesignatures/timesignaturesettingsmodel.cpp [hunk 2]
```diff
@@ -38,8 +38,13 @@ TimeSignatureSettingsModel::TimeSignatureSettingsModel(QObject* parent, IElement
...
+    m_horizontalScale = buildPropertyItem(mu::engraving::Pid::SCALE,
+                                          [this](const mu::engraving::Pid pid, const QVariant& newValue) {
         onPropertyValueChanged(pid, QSizeF(newValue.toDouble() / 100, m_verticalScale->value().toDouble() / 100));
+    },
+                                          [this](const mu::engraving::Sid sid, const QVariant& newValue) {
+        updateStyleValue(sid, QSizeF(newValue.toDouble() / 100, m_verticalScale->value().toDouble() / 100));
+        emit requestReloadPropertyItems();
     });
 
     m_verticalScale = buildPropertyItem(mu::engraving::Pid::SCALE, [this](const mu::engraving::Pid pid, const QVariant& newValue) {
```
Comment (cbjeukendrup) [inline src/inspector/models/notation/timesignatures/timesignaturesettingsmodel.cpp:50]: It looks like the vertical component of the property needs this callback too

Comment (mike-spa) [reply src/inspector/models/notation/timesignatures/timesignaturesettingsmodel.cpp]: aldo done 👍

**Review by cbjeukendrup — APPROVED** [review]

**Pull request merged**